lib/cdo/cloud_formation/cdo_app.rb

```diff
@@ -24,6 +24,7 @@ class CdoApp < StackTemplate
     end
 
     # Hard-coded constants and default values.
+    CHEF_CLIENT_BIN = '/usr/local/bin/chef-client'
```

Comment:
Can we name both the variable and the file itself something that indicates it's not just another copy of `chef-client`? It's clear from this PR, but confusing if you see it in isolation. Maybe something like `chef-client-with-secrets`?

Reply:
How about `chef-cdo-app`, to make it clear it's invoking `chef` to provision `cdo-app` (see 60b476c38b82a6ca769319a1e467ac3d556e19f0).